dotnet/src/Azure.Iot.Operations.Services/AssetAndDeviceRegistry/AdrServiceClient.cs

```diff
@@ -0,0 +1,222 @@
+using AssetEndpointProfile = Azure.Iot.Operations.Services.AssetAndDeviceRegistry.Models.AssetEndpointProfile;
+using NotificationResponse = Azure.Iot.Operations.Services.AssetAndDeviceRegistry.Models.NotificationResponse;
+
+namespace Azure.Iot.Operations.Services.AssetAndDeviceRegistry;
+
+public class AdrServiceClient(ApplicationContext applicationContext, IMqttPubSubClient mqttClient) : IAdrServiceClient
+{
+    private const string _aepNameTokenKey = "aepName";
+    private const byte _dummyByte = 1;
+    private static readonly TimeSpan _defaultTimeout = TimeSpan.FromSeconds(10);
+    private readonly AssetEndpointProfileServiceClientStub _assetEndpointProfileServiceClient = new(applicationContext, mqttClient);
+    private readonly AssetServiceClientStub _assetServiceClient = new(applicationContext, mqttClient);
+    private bool _disposed;
+    private readonly ConcurrentDictionary<string, byte> _observedAeps = new();
+    private readonly ConcurrentDictionary<string, byte> _observedAssets = new();
+
+    /// <inheritdoc/>
+    public async ValueTask DisposeAsync()
+    {
+        if (_disposed) return;
+        await _assetServiceClient.DisposeAsync().ConfigureAwait(false);
+        await _assetEndpointProfileServiceClient.DisposeAsync().ConfigureAwait(false);
+        GC.SuppressFinalize(this);
+        _disposed = true;
+    }
+
+    /// <inheritdoc/>
+    public async Task<NotificationResponse> ObserveAssetEndpointProfileUpdatesAsync(string aepName, TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        ObjectDisposedException.ThrowIf(_disposed, this);
+
+        _observedAeps[aepName] = _dummyByte;
+        await _assetServiceClient.AssetEndpointProfileUpdateEventTelemetryReceiver.StartAsync(cancellationToken);
+
+        var additionalTopicTokenMap = new Dictionary<string, string> { { _aepNameTokenKey, aepName } };
+        var notificationRequest = new NotifyOnAssetEndpointProfileUpdateRequestPayload
+        {
+            NotificationRequest = NotificationMessageType.On
+        };
+
+        var result = await _assetServiceClient.NotifyOnAssetEndpointProfileUpdateAsync(notificationRequest, null, additionalTopicTokenMap,
+            commandTimeout ?? _defaultTimeout, cancellationToken);
+        return result.NotificationResponse.ToModel();
+    }
+
+    /// <inheritdoc/>
+    public async Task<NotificationResponse> UnobserveAssetEndpointProfileUpdatesAsync(string aepName, TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        ObjectDisposedException.ThrowIf(_disposed, this);
+
+        if (_observedAeps.TryRemove(aepName, out _) && _observedAeps.IsEmpty)
+            await _assetServiceClient.AssetEndpointProfileUpdateEventTelemetryReceiver.StopAsync(cancellationToken);
```

Comment:
```suggestion
        if (_observedAeps.TryRemove(aepName, out _) && _observedAeps.IsEmpty)
        {
            await _assetServiceClient.AssetEndpointProfileUpdateEventTelemetryReceiver.StopAsync(cancellationToken);
        }
```

Try to avoid skipping braces for even one line if statements